### PR TITLE
Subscriptions: Add metering support

### PR DESCRIPTION
**Pull request by prateekbh**
- implements Metering
- adds metering to first level at rendering data

### CODE REVIEW

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ import {tryParseJson} from '../../../src/json';
 /** @const */
 const TAG = 'amp-subscriptions';
 
-/** @typedef {{loggedIn: boolean, subscribed: boolean, granted: boolean, entitlement: !JsonObject}} */
+/** @typedef {{loggedIn: boolean, subscribed: boolean, granted: boolean, entitlement: !JsonObject, metering: ./entitlement.MeteringData}} */
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Let's replace `metering:Array` to `metered:boolean` more in keeping with this structure.

File: extensions/amp-subscriptions/0.1/entitlement.js [hunk 2]
```diff
@@ -27,25 +30,29 @@ export class Entitlement {
...
    */
   static empty(service) {
-    return new Entitlement(
-        /* source */ '',
-        /* raw */ '',
-        service,
-        /* products */ [],
-        /* subscriptionToken */ null,
-        /* loggedIn */ false);
+    return new Entitlement({
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/entitlement.js]: Use as much as default state in this call as possible.

File: extensions/amp-subscriptions/0.1/entitlement.js [hunk 3]
```diff
@@ -58,7 +65,8 @@ export class Entitlement {
     this.subscriptionToken = subscriptionToken;
     /** @const {boolean} */
     this.loggedIn = loggedIn;
-
+    /** @const {MeteringData} */
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/entitlement.js]: Explicit `?`

File: extensions/amp-subscriptions/0.1/entitlement.js [hunk 4]
```diff
@@ -119,7 +128,18 @@ export class Entitlement {
     const products = json['products'] || [];
     const subscriptionToken = json['subscriptionToken'];
     const loggedIn = json['loggedIn'];
-    return new Entitlement(source, raw, /* service */ '',
-        products, subscriptionToken, loggedIn);
+    const meteringData = json['metering'];
+    let metering = null;
+    if (meteringData) {
+      metering = {
+        quotaLeft: meteringData['quotaLeft'],
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/entitlement.js]: What do you think about making them shorter? `left`, `total` still seem very readable to me.

File: extensions/amp-subscriptions/0.1/entitlement.js [hunk 5]
```diff
@@ -119,7 +128,18 @@ export class Entitlement {
...
+        quotaTotal: meteringData['quotaTotal'],
+        durationLeft: meteringData['durationLeft'],
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/entitlement.js]: This should be simply a `resetTime` with a time_t value.

File: extensions/amp-subscriptions/0.1/entitlement.js (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/entitlement.js]: Done

File: extensions/amp-subscriptions/0.1/entitlement.js (hunk 3 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/entitlement.js]: Done

File: extensions/amp-subscriptions/0.1/entitlement.js (hunk 4 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/entitlement.js]: Done

File: extensions/amp-subscriptions/0.1/entitlement.js (hunk 5 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/entitlement.js]: Done

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js (hunk 1 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Done

**Pull request merged**